docs/CONFIG.md

```diff
@@ -0,0 +1,96 @@
+# Config System Structure
+
+There are 4 basic component types under `config/_base_`, dataset, model, schedule, default\*runtime.
+Many method could be easily constructed with one of each like Faster R-CNN, Mask R-CNN, Cascade R-CNN, RPN, SSD.
+The configs that are composed by components from `_base_` are called _primitive_.
+
+For all configs under the same folder, it is recommended to have only **one** _primitive_ config. All other configs should inherit from the _primitive_ config. In this way, the maximum of inheritance level is 3.
+
+For easy understanding, we recommend contributors to inherit from exiting methods.
+For example, if some modification is made base on Faster R-CNN, user may first inherit the basic Faster R-CNN structure by specifying `_base_ = ../faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py`, then modify the necessary fields in the config files.
+
+If you are building an entirely new method that does not share the structure with any of the existing methods, you may create a folder `xxx_rcnn` under `configs`,
+
+Please refer [here](https://mmcv.readthedocs.io/en/latest/utils.html#config) for detailed documentation.
+
+# Config Name Style
+
+We follow the below style to name config files. Contributors are advised to follow the same style.
+
+```
+[model]_(model setting)_[backbone]_[neck]_(norm setting)_(misc)_(gpu x batch)_[schedule]_[dataset]
+```
+
+`[xxx]` is required field and `(yyy)` is optional.
```

Comment:
`[]` is usually used to indicate optional. I recommend using `{}` and `[]` respectively.